pkg/services/ngalert/notifier/channels/discord_test.go

```diff
@@ -58,6 +57,70 @@ func TestDiscordNotifier(t *testing.T) {
 			},
 			expMsgError: nil,
 		},
+		{
+			name: "Missing field in template",
+			settings: `{
+				"avatar_url": "https://grafana.com/assets/img/fav32.png",
+				"url": "http://localhost",
+				"message": "I'm a custom template {{ .NotAField }} bad template"
+			}`,
+			alerts: []*types.Alert{
+				{
+					Alert: model.Alert{
+						Labels:      model.LabelSet{"alertname": "alert1", "lbl1": "val1"},
+						Annotations: model.LabelSet{"ann1": "annv1"},
+					},
+				},
+			},
+			expMsg: map[string]interface{}{
+				"avatar_url": "https://grafana.com/assets/img/fav32.png",
+				"content":    "I'm a custom template ",
+				"embeds": []interface{}{map[string]interface{}{
+					"color": 1.4037554e+07,
+					"footer": map[string]interface{}{
+						"icon_url": "https://grafana.com/assets/img/fav32.png",
+						"text":     "Grafana v" + setting.BuildVersion,
+					},
+					"title": "[FIRING:1]  (val1)",
+					"url":   "http://localhost/alerting/list",
+					"type":  "rich",
+				}},
+				"username": "Grafana",
+			},
+			expMsgError: nil,
+		},
+		{
+			name: "Invalid template",
```

Comment:
The PR fixes 3 distinct error cases:
- when the template for `content` fails (should truncate and affect no other fields)
- when the template for `avatar_url` fails (should replace with default avatar and affect no other fields)
- when the template for `url` fails (should replace with default url)

But, the added tests only cover the first one - the tests only cover invalid templates in `content`. Could we please add additional test cases, using an invalid `avatar_url` and an invalid `url` template?